backend/src/LibExecution/Interpreter.fs

```diff
@@ -425,7 +431,10 @@ let execute (exeState : ExecutionState) (vm : VMState) : Ply<Dval> =
 
         | CreateLambda(lambdaReg, impl) ->
           vm.lambdaInstrCache <-
-            Map.add (currentFrame.context, impl.exprId) impl vm.lambdaInstrCache
+            vm.lambdaInstrCache
+            |> Map.add (currentFrame.context, impl.exprId) impl
+            |> Map.add (Source, impl.exprId) impl
```

Comment:
I suppose this is a similar question - why do we need to add two entries?

Reply:
the lambda is created in the `Source` context, but we try to apply it in the `PackageFn` context 
I've added this to make sure that we can find the lambda later regardless of the context in which it's being applied. Again not sure if it is the right thing to do, please let me know if you think there is a better way to do it